Specs/BSPropertySpec.mm

```diff
@@ -33,10 +41,24 @@
             });
         });
 
-        context(@"when the return type is a non-existent objective-c class", ^{
+        context(@"when the return type is an objective-c class conforming to a protocol", ^{
+            it(@"determines the protocol", ^{
+                property = [BSProperty propertyWithClass:[ClassWithProtocolInstance class] propertyNameString:@"protocolInstance"];
+                expect(property.returnType == [TestProtocolImpl class]).to(equal(YES));
+            });
+        });
+        
+        context(@"when the return type is an objective-c protocol", ^{
+            it(@"determines the protocol", ^{
+                property = [BSProperty propertyWithClass:[ClassWithProtocolProperty class] propertyNameString:@"protocolObject"];
+                expect(property.returnType == @protocol(TestProtocol)).to(equal(YES));
+            });
+        });
+        
+        context(@"when the return type conforms to multiple objective-c protocols", ^{
```

Comment:
Thanks for your work here. I'm inclined to merge it in later today, and I've asked a couple projects to try it out. Really appreciate the contribution!

On Mon, Nov 30, 2015 at 7:55 AM, xtreme-steven-wu
<notifications@github.com> wrote:

>> -        context(@"when the return type is a non-existent objective-c class", ^{
>> +        context(@"when the return type is an objective-c class conforming to a protocol", ^{
>> +            it(@"determines the protocol", ^{
>> +                property = [BSProperty propertyWithClass:[ClassWithProtocolInstance class] propertyNameString:@"protocolInstance"];
>> +                expect(property.returnType == [TestProtocolImpl class]).to(equal(YES));
>> +            });
>> +        });
>> +        
>> +        context(@"when the return type is an objective-c protocol", ^{
>> +            it(@"determines the protocol", ^{
>> +                property = [BSProperty propertyWithClass:[ClassWithProtocolProperty class] propertyNameString:@"protocolObject"];
>> +                expect(property.returnType == @protocol(TestProtocol)).to(equal(YES));
>> +            });
>> +        });
>> +        
>> +        context(@"when the return type conforms to multiple objective-c protocols", ^{
> Bump, thoughts about merging this in?
> Don't think protocol injects are working at all at the moment.
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/jbsf/blindside/pull/47/files#r46160887